src/Features/LanguageServer/Protocol/Handler/RequestExecutionQueue.cs

```diff
@@ -145,164 +160,115 @@ public void Shutdown()
             CancellationToken requestCancellationToken)
             where TRequestType : class
         {
-            // Create a task completion source that will represent the processing of this request to the caller
-            var completion = new TaskCompletionSource<TResponseType?>();
-
             // Note: If the queue is not accepting any more items then TryEnqueue below will fail.
 
             var textDocument = handler.GetTextDocumentIdentifier(request);
-            var item = new QueueItem(
+
+            // Create a combined cancellation token so either the client cancelling it's token or the queue
+            // shutting down cancels the request.
+            using var combinedTokenSource = _cancelSource.Token.CombineWith(requestCancellationToken);
```

Comment:
do you actually want the `using` here?  won't that dispose this at the end of this scope?